apps/judicial-system/backend/src/app/modules/case/test/archive.spec.ts

```diff
@@ -0,0 +1,403 @@
+interface Then {
+  result: ArchiveResponse
+  error: Error
+}
+
+type GivenWhenThen = () => Promise<Then>
+
+describe('CaseController - Archive', () => {
+  let mockFileService: FileService
+  let mockDefendantService: DefendantService
+  let mockCaseModel: typeof Case
+  let mockCaseArchiveModel: typeof CaseArchive
+  let mockCaseConfig: ConfigType<typeof caseModuleConfig>
+  let transaction: Transaction
+  let givenWhenThen: GivenWhenThen
+
+  beforeEach(async () => {
+    const {
+      fileService,
+      defendantService,
+      sequelize,
+      caseModel,
+      caseArchiveModel,
+      caseConfig,
+      caseController,
+    } = await createTestingCaseModule()
+
+    mockFileService = fileService
+    mockDefendantService = defendantService
+    mockCaseModel = caseModel
+    mockCaseArchiveModel = caseArchiveModel
+    mockCaseConfig = caseConfig
+
+    const mockTransaction = sequelize.transaction as jest.Mock
+    transaction = {} as Transaction
+    mockTransaction.mockImplementationOnce(
+      (fn: (transaction: Transaction) => unknown) => fn(transaction),
+    )
+
+    givenWhenThen = async () => {
+      const then = {} as Then
+
+      await caseController
+        .archive()
+        .then((res) => (then.result = res))
+        .catch((error) => (then.error = error))
+
+      return then
+    }
+  })
+
+  describe('case lookup', () => {
+    beforeEach(async () => {
+      await givenWhenThen()
+    })
+
+    it('should lookup a case', () => {
+      expect(mockCaseModel.findOne).toHaveBeenCalledWith({
+        include: [
+          { model: Defendant, as: 'defendants' },
+          { model: Institution, as: 'court' },
+          {
+            model: User,
+            as: 'creatingProsecutor',
+            include: [{ model: Institution, as: 'institution' }],
+          },
+          {
+            model: User,
+            as: 'prosecutor',
+            include: [{ model: Institution, as: 'institution' }],
+          },
+          { model: Institution, as: 'sharedWithProsecutorsOffice' },
+          {
+            model: User,
+            as: 'judge',
+            include: [{ model: Institution, as: 'institution' }],
+          },
+          {
+            model: User,
+            as: 'registrar',
+            include: [{ model: Institution, as: 'institution' }],
+          },
+          {
+            model: User,
+            as: 'courtRecordSignatory',
+            include: [{ model: Institution, as: 'institution' }],
+          },
+          { model: Case, as: 'parentCase' },
+          { model: Case, as: 'childCase' },
+          { model: CaseFile, as: 'caseFiles' },
+        ],
+        order: [
+          [{ model: Defendant, as: 'defendants' }, 'created', 'ASC'],
+          [{ model: CaseFile, as: 'caseFiles' }, 'created', 'ASC'],
+        ],
+        where: {
+          isArchived: false,
+          [Op.or]: [
+            { state: CaseState.DELETED },
+            {
+              [Op.or]: [
+                {
```

Comment:
I don't really see the value in checking if `caseModel.fineOne` has been called with the right parameters. Can we test the return value? 
If we are not able to test the return value, can we reuse the `oldFilter` here?

Reply:
I am mocking the return value and that is the reason I test the call parameters. We have no way of testing the return value unless we call a running database and that involves a pretty hefty test setup cost.

Basically, I am saying: If our code makes the correct call, then I trust the database to return the expected value.

Using oldFilter in the test is a good idea.

Reply:
I see, then this makes sense.